src/components/Navigation/Pages/PageDefinitions/PageDefinitions.js

```diff
@@ -97,11 +97,51 @@ export class PageDefinitions extends UserManagedList {
     )
   }
 
+  toggleCheckbox = index => {
```

Comment:
I suggest to` move toggleCheckbox , onSelectAll, onFieldChange,  resetComponents`,  methods to the User Managed List component.